### PR TITLE
Fix paired end bug

### PR DESCRIPTION
**Pull request by mirpedrol**
When I introduced the ability to use the meta map key `single_end` I changed the value for some channels. This made it impossible to join those channels with the original ones as they had different metas.
This PR removes the `single_end` from the meta map to be able to join the channels. It also changes the value from reference, template, and protospacer channels accordingly.

### CODE REVIEW

**Review by LaurenceKuhl — APPROVED** [review]

**Comment by LaurenceKuhl** [comment]
LGTM but maybe you could comment the channel magic a little bit more? :D

File: workflows/crisprseq_targeted.nf [hunk 1]
```diff
@@ -223,25 +264,48 @@ workflow CRISPRSEQ_TARGETED {
         ch_cat_fastq.paired
             .mix(ch_cat_fastq.single)
             .join(PEAR.out.assembled, remainder: true)
-            .join(SEQTK_SEQ_MASK.out.fastx)
-            .join(CUTADAPT.out.log)
-            .map { meta, reads, assembled, masked, trimmed ->
+            .map { meta, reads, assembled ->
+                return [ meta - meta.subMap('single_end'), reads, assembled]
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:268]:
```suggestion
                // Remove the single_end key from the meta map to allow joining channels with different single_end values
                return [ meta - meta.subMap('single_end'), reads, assembled]
```

File: workflows/crisprseq_targeted.nf [hunk 2]
```diff
@@ -223,25 +264,48 @@ workflow CRISPRSEQ_TARGETED {
...
             }
             .set { ch_preprocessing_summary_data }
     } else {
         ch_cat_fastq.paired
             .mix(ch_cat_fastq.single)
             .join(PEAR.out.assembled, remainder: true)
-            .join(SEQTK_SEQ_MASK.out.fastx)
-            .map { meta, reads, assembled, masked ->
+            .map { meta, reads, assembled ->
+                return [ meta - meta.subMap('single_end'), reads, assembled]
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:295]:
```suggestion
                // Remove the single_end key from the meta map to allow joining channels with different single_end values
                return [ meta - meta.subMap('single_end'), reads, assembled]
```

File: workflows/crisprseq_targeted.nf [hunk 3]
```diff
@@ -157,6 +157,47 @@ workflow CRISPRSEQ_TARGETED {
     .set { ch_pear_fastq }
     ch_versions = ch_versions.mix(PEAR.out.versions)
 
+    // Change reference, protospacer and template channels to have the same meta information as the reads
+    ch_pear_fastq
+        .map {meta, reads ->
+            single_end = meta.single_end
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:163]:
```suggestion
            // save single_end value and remove the key from the meta map
            single_end = meta.single_end
```

File: workflows/crisprseq_targeted.nf [hunk 4]
```diff
@@ -157,6 +157,47 @@ workflow CRISPRSEQ_TARGETED {
...
+            return [ meta - meta.subMap('single_end'), reads, single_end ]
+        }
+        .tap { no_single_end }
+        .join(
+            ORIENT_REFERENCE.out.reference
+            .map {meta, reference ->
+                return [ meta - meta.subMap('single_end'), reference ]
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:170]:
```suggestion
                // Remove single_end from the meta map to allow joining two channels with different single_end values
                return [ meta - meta.subMap('single_end'), reference ]
```

File: workflows/crisprseq_targeted.nf [hunk 5]
```diff
@@ -157,6 +157,47 @@ workflow CRISPRSEQ_TARGETED {
...
+            }
+        )
+        .map {meta, reads, single_end, reference ->
+            return [ meta + ["single_end": single_end], reference ]
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:174]:
```suggestion
            // Add the correct single_end value to the reference meta map.
            return [ meta + ["single_end": single_end], reference ]
```

**Pull request merged**